quantus/helpers/model/pytorch_model.py

```diff
@@ -219,3 +219,67 @@ def sample(
                             "or 'additive' (string) when you sample the model."
                         )
         return model_copy
+
+    def get_hidden_representations(
+        self,
+        x: np.ndarray,
+        layer_names: Optional[List[str]] = None,
+        layer_indices: Optional[List[int]] = None,
+    ) -> np.ndarray:
+
+        device = self.device if self.device is not None else "cpu"
+        all_layers = [*self.model.named_modules()]
+        num_layers = len(all_layers)
+
+        if layer_indices is None:
+            layer_indices = []
+
+        # Convert negative indices to positive
+        positive_layer_indices = [
+            i if i >= 0 else num_layers + i for i in layer_indices
+        ]
+
+        if layer_names is None:
+            layer_names = []
+
+        def is_layer_of_interest(index, name):
+            if layer_names == [] and positive_layer_indices == []:
+                return True
+            return index in positive_layer_indices or name in layer_names
+
+        # skip modules defined by subclassing API
```

Comment:
Formatting of comments... please check everywhere.

Reply:
done